homeassistant/components/slack/notify.py

```diff
@@ -193,14 +193,22 @@ async def _async_send_remote_file_message(
 
     async def _async_send_text_only_message(self, targets, message, title, blocks):
         """Send a text-only message."""
+        username = self._username
```

Comment:
Because of the dereferencing of the dictionary needed for the dynamic parameter name, the context of `self` becomes the slack message client. Setting the class variable to a local variable allows a proper closure to be formed when passing it to `chat_postMessage()`

Reply:
I'd be thrilled if you have a better way of doing this, but I was trying to avoid having to create two copies of the call to `chat_postMessage()` with the only difference between them being the `icon_*` parameter.